File: common/src/main/java/wile/redstonepen/libmc/IPlatformHelper.java
package wile.redstonepen.libmc;

import net.minecraft.server.level.ServerLevel;
import net.minecraft.world.entity.player.Player;
import java.nio.file.Path;
import java.util.Optional;

public interface IPlatformHelper
{
  Path getGameDirectory();
  boolean isModLoaded(String modId);
  Optional<? extends Player> getFakePlayer(ServerLevel world);
}

File: common/src/main/kotlin/wile/redstonepen/platform/IPlatformHelper.kt
package wile.redstonepen.platform

import java.nio.file.Path
import java.util.Optional
import net.minecraft.server.level.ServerLevel
import net.minecraft.world.entity.player.Player

interface IPlatformHelper {
    fun getGameDirectory(): Path

    fun isModLoaded(modId: String): Boolean

    fun getFakePlayer(world: ServerLevel): Optional<out Player>
}
